node/repo/fsrepo.go

```diff
@@ -41,47 +41,83 @@ const (
 	fsKeystore      = "keystore"
 )
 
-type RepoType int
+func NewRepoTypeFromString(t string) RepoType {
+	switch t {
+	case "FullNode":
+		return FullNodeRepoType{}
+	case "StorageMiner":
+		return StorageMinerRepoType{}
+	case "Worker":
+		return WorkerRepoType{}
+	case "Wallet":
+		return WalletRepoType{}
+	default:
+		panic("unknown RepoType")
+	}
+}
 
-const (
-	_                 = iota // Default is invalid
-	FullNode RepoType = iota
-	StorageMiner
-	Worker
-	Wallet
-	Markets
-)
+type RepoType interface {
+	Type() string
+	Config() interface{}
+}
 
-func (t RepoType) String() string {
-	s := [...]string{
-		"__invalid__",
-		"FullNode",
-		"StorageMiner",
-		"Worker",
-		"Wallet",
-		"Markets",
-	}
-	if t < 0 || int(t) > len(s) {
-		return "__invalid__"
-	}
-	return s[t]
+type FullNodeRepoType struct {
```

Comment:
The `RepoType` suffix feels a bit redundant.

Reply:
Also given that all of those structs are field-less it could be nicer to do

```
type fullNode struct {}
var FullNode fullNode
```

This way we don't need to add `{}` everywhere